[Bug] Can not sink using Flink connector when StarRocks deployed in `shared_data` mode

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. deploy starrocks in `shared_data` mode
2. create table 
3. write data using Flink connector

### Expected behavior (Required)
> data can sink to starrocks 

```result
StarRocks > SHOW PROC '/backends';
Empty set (0.00 sec)

StarRocks > show proc '/compute_nodes';
+---------------+----------------------------------------------------------------+---------------+--------+----------+----------+---------------------+---------------------+-------+----------------------+-----------------------+--------+------------------+----------+-------------------+------------+------------+----------------+-------------+----------+
| ComputeNodeId | IP                                                             | HeartbeatPort | BePort | HttpPort | BrpcPort | LastStartTime       | LastHeartbeat       | Alive | SystemDecommissioned | ClusterDecommissioned | ErrMsg | Version          | CpuCores | NumRunningQueries | MemUsedPct | CpuUsedPct | HasStoragePath | StarletPort | WorkerId |
+---------------+----------------------------------------------------------------+---------------+--------+----------+----------+---------------------+---------------------+-------+----------------------+-----------------------+--------+------------------+----------+-------------------+------------+------------+----------------+-------------+----------+
| 10004         | starrocks-be-s3-0.starrocks-be-s3.olap.svc.cluster.local | 9050          | 9060   | 8040     | 8060     | 2023-10-20 17:10:22 | 2023-10-22 15:44:51 | true  | false                | false                 |        | 3.1.2-4f3a2ee91b | 2        | 0                 | 0.86 %     | 0.0 %      | 9070           | 1           | true     |
| 10030         | starrocks-be-s3-1.starrocks-be-s3.olap.svc.cluster.local | 9050          | 9060   | 8040     | 8060     | 2023-10-20 17:12:12 | 2023-10-22 15:44:51 | true  | false                | false                 |        | 3.1.2-4f3a2ee91b | 2        | 0                 | 0.88 %     | 0.0 %      | 9070           | 2           | true     |
| 10055         | starrocks-be-s3-2.starrocks-be-s3.olap.svc.cluster.local | 9050          | 9060   | 8040     | 8060     | 2023-10-20 17:13:32 | 2023-10-22 15:44:51 | true  | false                | false                 |        | 3.1.2-4f3a2ee91b | 2        | 0                 | 1.37 %     | 0.0 %      | 9070           | 3           | true     |
+---------------+----------------------------------------------------------------+---------------+--------+----------+----------+---------------------+---------------------+-------+----------------------+-----------------------+--------+------------------+----------+-------------------+------------+------------+----------------+-------------+----------+
3 rows in set (0.00 sec)
```

### Real behavior (Required)
> log stack trace

```log
java.lang.RuntimeException: com.starrocks.data.load.stream.exception.StreamLoadFailException: Transaction start failed, db: xxx, table: xxx, label: flink-ea913273-e5fb-4259-9c66-16b680409de2, responseBody: {
  "Status": "FAILED",
  "Message": "class com.starrocks.common.UserException: No backend alive."
}
	at com.starrocks.data.load.stream.v2.StreamLoadManagerV2.AssertNotException(StreamLoadManagerV2.java:427) ~[?:?]
	at com.starrocks.data.load.stream.v2.StreamLoadManagerV2.write(StreamLoadManagerV2.java:252) ~[?:?]
	at com.starrocks.connector.flink.table.sink.StarRocksDynamicSinkFunctionV2.invoke(StarRocksDynamicSinkFunctionV2.java:197) ~[?:?]
	at org.apache.flink.streaming.api.operators.StreamSink.processElement(StreamSink.java:54) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:71) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:46) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:26) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:50) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:28) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.api.operators.StreamMap.processElement(StreamMap.java:38) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:71) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:46) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:26) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:50) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:28) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$ManualWatermarkContext.processAndCollect(StreamSourceContexts.java:317) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.api.operators.StreamSourceContexts$WatermarkContext.collect(StreamSourceContexts.java:411) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at com.kedacom.starrocks.source.DeviceSource.run(DeviceSource.java:35) ~[?:?]
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:104) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:60) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.SourceStreamTask$LegacySourceFunctionThread.run(SourceStreamTask.java:269) ~[flink-dist_2.12-1.13-SNAPSHOT.jar:1.13-SNAPSHOT]
Caused by: com.starrocks.data.load.stream.exception.StreamLoadFailException: Transaction start failed, db: xxx, table: xxx, label: flink-ea913273-e5fb-4259-9c66-16b680409de2, responseBody: {
  "Status": "FAILED",
  "Message": "class com.starrocks.common.UserException: No backend alive."
}
	at com.starrocks.data.load.stream.TransactionStreamLoader.doBegin(TransactionStreamLoader.java:153) ~[?:?]
	at com.starrocks.data.load.stream.TransactionStreamLoader.begin(TransactionStreamLoader.java:99) ~[?:?]
	at com.starrocks.data.load.stream.DefaultStreamLoader.send(DefaultStreamLoader.java:170) ~[?:?]
	at com.starrocks.data.load.stream.v2.TransactionTableRegion.streamLoad(TransactionTableRegion.java:331) ~[?:?]
	at com.starrocks.data.load.stream.v2.TransactionTableRegion.flush(TransactionTableRegion.java:228) ~[?:?]
	at com.starrocks.data.load.stream.v2.StreamLoadManagerV2.lambda$init$0(StreamLoadManagerV2.java:220) ~[?:?]
	at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_362]
```
### StarRocks version (Required)
 - 3.1.2

--- a/fe/fe-core/src/main/java/com/starrocks/http/rest/TransactionLoadAction.java
+++ b/fe/fe-core/src/main/java/com/starrocks/http/rest/TransactionLoadAction.java
@@ -45,7 +45,8 @@ import com.starrocks.http.BaseRequest;
 import com.starrocks.http.BaseResponse;
 import com.starrocks.http.IllegalArgException;
 import com.starrocks.server.GlobalStateMgr;
-import com.starrocks.system.Backend;
+import com.starrocks.server.RunMode;
+import com.starrocks.system.ComputeNode;
 import com.starrocks.thrift.TNetworkAddress;
 import com.starrocks.transaction.TransactionStatus;
 import io.netty.handler.codec.http.HttpMethod;
@@ -70,9 +71,10 @@ public class TransactionLoadAction extends RestBaseAction {
     private static final String CHANNEL_ID_STR = "channel_id";
     private static TransactionLoadAction ac;
 
-    private Map<String, Long> txnBackendMap = new LinkedHashMap<String, Long>(512, 0.75f, true) {
+    private Map<String, Long> txnNodeMap = new LinkedHashMap<String, Long>(512, 0.75f, true) {
         protected boolean removeEldestEntry(Map.Entry<String, Long> eldest) {
-            return size() > GlobalStateMgr.getCurrentSystemInfo().getTotalBackendNumber() * 512;
+            return size() > (GlobalStateMgr.getCurrentSystemInfo().getTotalBackendNumber() +
+                    GlobalStateMgr.getCurrentSystemInfo().getTotalComputeNodeNumber()) * 512;
         }
     };
 
@@ -80,9 +82,9 @@ public class TransactionLoadAction extends RestBaseAction {
         super(controller);
     }
 
-    public int txnBackendMapSize() {
+    public int txnNodeMapSize() {
         synchronized (this) {
-            return txnBackendMap.size();
+            return txnNodeMap.size();
         }
     }
 
@@ -141,7 +143,7 @@ public class TransactionLoadAction extends RestBaseAction {
             throw new DdlException("Must provide channel_num when stream load begin.");
         }
 
-        Long backendID = null;
+        Long nodeID = null;
 
         if (Strings.isNullOrEmpty(dbName)) {
             throw new UserException("No database selected.");
@@ -206,15 +208,11 @@ public class TransactionLoadAction extends RestBaseAction {
             synchronized (this) {
                 // 2.1 save label->be map when begin transaction, so that subsequent operator can send to same BE
                 if (op.equalsIgnoreCase(TXN_BEGIN)) {
-                    List<Long> backendIds = GlobalStateMgr.getCurrentSystemInfo().seqChooseBackendIds(1, true, false);
-                    if (CollectionUtils.isEmpty(backendIds)) {
-                        throw new UserException("No backend alive.");
-                    }
-                    backendID = backendIds.get(0);
-                    // txnBackendMap is LRU cache, it automic remove unused entry
-                    txnBackendMap.put(label, backendID);
+                    nodeID = getBackendOrComputeId();
+                    // txnNodeMap is LRU cache, it atomic remove unused entry
+                    txnNodeMap.put(label, nodeID);
                 } else if (channelIdStr == null) {
-                    backendID = txnBackendMap.get(label);
+                    nodeID = txnNodeMap.get(label);
                 }
             }
         }
@@ -268,7 +266,6 @@ public class TransactionLoadAction extends RestBaseAction {
         }
 
         if (op.equalsIgnoreCase(TXN_COMMIT) && channelIdStr != null) {
-            int channelId = Integer.parseInt(channelIdStr);
             TransactionResult resp = new TransactionResult();
             GlobalStateMgr.getCurrentState().getStreamLoadMgr().commitLoadTask(label, resp);
             sendResult(request, response, resp);
@@ -276,7 +273,6 @@ public class TransactionLoadAction extends RestBaseAction {
         }
 
         if (op.equalsIgnoreCase(TXN_ROLLBACK) && channelIdStr != null) {
-            int channelId = Integer.parseInt(channelIdStr);
             TransactionResult resp = new TransactionResult();
             GlobalStateMgr.getCurrentState().getStreamLoadMgr().rollbackLoadTask(label, resp);
             sendResult(request, response, resp);
@@ -284,20 +280,39 @@ public class TransactionLoadAction extends RestBaseAction {
         }
 
 
-        if (backendID == null) {
-            throw new UserException("transaction with op " + op + " label " + label + " has no backend");
+        if (nodeID == null) {
+            throw new UserException("transaction with op " + op + " label " + label + " has no node");
         }
 
-        Backend backend = GlobalStateMgr.getCurrentSystemInfo().getBackend(backendID);
-        if (backend == null) {
-            throw new UserException("Backend " + backendID + " is not alive");
+        ComputeNode node = GlobalStateMgr.getCurrentSystemInfo().getBackend(nodeID);
+        if (node == null) {
+            node = GlobalStateMgr.getCurrentSystemInfo().getComputeNode(nodeID);
+            if (node == null) {
+                throw new UserException("Node " + nodeID + " is not alive");
+            }
         }
 
-        TNetworkAddress redirectAddr = new TNetworkAddress(backend.getHost(), backend.getHttpPort());
+        TNetworkAddress redirectAddr = new TNetworkAddress(node.getHost(), node.getHttpPort());
 
         LOG.info("redirect transaction action to destination={}, db: {}, table: {}, op: {}, label: {}",
                 redirectAddr, dbName, tableName, op, label);
         redirectTo(request, response, redirectAddr);
+    }
+
+    private static Long getBackendOrComputeId() throws UserException {
+        List<Long> backendIds = GlobalStateMgr.getCurrentSystemInfo().seqChooseBackendIds(1, true, false);
+        if (CollectionUtils.isNotEmpty(backendIds)) {
+            return backendIds.get(0);
+        }
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_NOTHING) {
+            throw new UserException("No backend alive.");
+        }
+        List<Long> computeNodes = GlobalStateMgr.getCurrentSystemInfo().seqChooseComputeNodes(1, true,
+                false);
+        if (CollectionUtils.isNotEmpty(computeNodes)) {
+            return computeNodes.get(0);
+        }
+        throw new UserException("No backend or compute node alive.");
     }
 }
 

--- a/fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java
+++ b/fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java
@@ -106,8 +106,8 @@ public class SystemInfoService implements GsonPostProcessable {
     @SerializedName(value = "ce")
     private volatile ConcurrentHashMap<Long, ComputeNode> idToComputeNodeRef;
 
-    private long lastBackendIdForCreation = -1;
-    private long lastBackendIdForOther = -1;
+    private long lastNodeIdForCreation = -1;
+    private long lastNodeIdForOther = -1;
 
     private volatile ImmutableMap<Long, AtomicLong> idToReportVersionRef;
     private volatile ImmutableMap<Long, DiskInfo> pathHashToDishInfoRef;
@@ -263,7 +263,7 @@ public class SystemInfoService implements GsonPostProcessable {
         String opMessage;
         formatSb.append("%s:%d's host has been modified to %s");
         if (candidateBackends.size() >= 2) {
-            formatSb.append("\nplease exectue %d times, to modify the remaining backends\n");
+            formatSb.append("\nplease execute %d times, to modify the remaining backends\n");
             for (int i = 1; i < candidateBackends.size(); i++) {
                 Backend be = candidateBackends.get(i);
                 formatSb.append(be.getHost() + ":" + be.getHeartbeatPort() + "\n");
@@ -615,6 +615,12 @@ public class SystemInfoService implements GsonPostProcessable {
         return idToBackendRef.size();
     }
 
+    public int getTotalComputeNodeNumber() {
+        return idToComputeNodeRef.size();
+    }
+
+
+
     public int getAliveComputeNodeNumber() {
         return getComputeNodeIds(true).size();
     }
@@ -705,12 +711,18 @@ public class SystemInfoService implements GsonPostProcessable {
 
     public List<Backend> getAvailableBackends() {
         return getBackends().stream()
-                .filter(v -> v.isAvailable())
+                .filter(ComputeNode::isAvailable)
                 .collect(Collectors.toList());
     }
 
     public List<ComputeNode> getComputeNodes() {
         return Lists.newArrayList(idToComputeNodeRef.values());
+    }
+
+    public List<ComputeNode> getAvailableComputeNodes() {
+        return getComputeNodes().stream()
+                .filter(ComputeNode::isAvailable)
+                .collect(Collectors.toList());
     }
 
     public Stream<ComputeNode> backendAndComputeNodeStream() {
@@ -729,16 +741,26 @@ public class SystemInfoService implements GsonPostProcessable {
         return seqChooseBackendIds(backendNum, needAvailable, isCreate, v -> !v.diskExceedLimit());
     }
 
+    public List<Long> seqChooseComputeNodes(int computeNodeNum, boolean needAvailable, boolean isCreate) {
+
+        final List<ComputeNode> candidateComputeNodes = needAvailable ? getAvailableComputeNodes() : getComputeNodes();
+        if (CollectionUtils.isEmpty(candidateComputeNodes)) {
+            LOG.warn("failed to find any compute nodes, needAvailable={}", needAvailable);
+            return Collections.emptyList();
+        }
+
+        return seqChooseNodeIds(computeNodeNum, isCreate, candidateComputeNodes);
+    }
+
     private List<Long> seqChooseBackendIds(int backendNum, boolean needAvailable, boolean isCreate,
                                            Predicate<? super Backend> predicate) {
-
         final List<Backend> candidateBackends = needAvailable ? getAvailableBackends() : getBackends();
         if (CollectionUtils.isEmpty(candidateBackends)) {
             LOG.warn("failed to find any backend, needAvailable={}", needAvailable);
             return Collections.emptyList();
         }
 
-        final List<Backend> filteredBackends = candidateBackends.stream()
+        final List<ComputeNode> filteredBackends = candidateBackends.stream()
                 .filter(predicate)
                 .collect(Collectors.toList());
 
@@ -752,66 +774,66 @@ public class SystemInfoService implements GsonPostProcessable {
                     candidateBackends.size(), needAvailable, backendInfo);
             return Collections.emptyList();
         }
-        return seqChooseBackendIds(backendNum, isCreate, filteredBackends);
+        return seqChooseNodeIds(backendNum, isCreate, filteredBackends);
     }
 
     /**
-     * choose backends by round-robin
+     * choose nodes by round-robin
      *
-     * @param backendNum  number of backend wanted
-     * @param isCreate    last backend id for creation
-     * @param srcBackends list of the candidate backends
-     * @return empty list if not enough backend, otherwise return a list of backend's id
+     * @param nodeNum  number of node wanted
+     * @param isCreate    last node id for creation
+     * @param srcNodes list of the candidate nodes
+     * @return empty list if not enough node, otherwise return a list of node's id
      */
-    public synchronized List<Long> seqChooseBackendIds(int backendNum, boolean isCreate, final List<Backend> srcBackends) {
+    public synchronized List<Long> seqChooseNodeIds(int nodeNum, boolean isCreate, final List<ComputeNode> srcNodes) {
 
-        long lastBackendId;
+        long lastNodeId;
 
         if (isCreate) {
-            lastBackendId = lastBackendIdForCreation;
+            lastNodeId = lastNodeIdForCreation;
         } else {
-            lastBackendId = lastBackendIdForOther;
+            lastNodeId = lastNodeIdForOther;
         }
 
         // host -> BE list
-        Map<String, List<Backend>> backendMaps = Maps.newHashMap();
-        for (Backend backend : srcBackends) {
-            String host = backend.getHost();
+        Map<String, List<ComputeNode>> nodeMaps = Maps.newHashMap();
+        for (ComputeNode node : srcNodes) {
+            String host = node.getHost();
 
-            if (!backendMaps.containsKey(host)) {
-                backendMaps.put(host, Lists.newArrayList());
+            if (!nodeMaps.containsKey(host)) {
+                nodeMaps.put(host, Lists.newArrayList());
             }
 
-            backendMaps.get(host).add(backend);
+            nodeMaps.get(host).add(node);
         }
 
         // if more than one backend exists in same host, select a backend at random
-        List<Backend> backends = Lists.newArrayList();
-        for (List<Backend> list : backendMaps.values()) {
+        List<ComputeNode> nodes = Lists.newArrayList();
+        for (List<ComputeNode> list : nodeMaps.values()) {
             Collections.shuffle(list);
-            backends.add(list.get(0));
+            nodes.add(list.get(0));
         }
 
-        List<Long> backendIds = Lists.newArrayList();
-        // get last backend index
-        int lastBackendIndex = -1;
+        List<Long> nodeIds = Lists.newArrayList();
+        // get last node index
+        int lastNodeIndex = -1;
         int index = -1;
-        for (Backend backend : backends) {
+        for (ComputeNode node : nodes) {
             index++;
-            if (backend.getId() == lastBackendId) {
-                lastBackendIndex = index;
+            if (node.getId() == lastNodeId) {
+                lastNodeIndex = index;
                 break;
             }
         }
-        Iterator<Backend> iterator = Iterators.cycle(backends);
+        Iterator<ComputeNode> iterator = Iterators.cycle(nodes);
         index = -1;
         boolean failed = false;
         // 2 cycle at most
-        int maxIndex = 2 * backends.size();
-        while (iterator.hasNext() && backendIds.size() < backendNum) {
-            Backend backend = iterator.next();
+        int maxIndex = 2 * nodes.size();
+        while (iterator.hasNext() && nodeIds.size() < nodeNum) {
+            ComputeNode node = iterator.next();
             index++;
-            if (index <= lastBackendIndex) {
+            if (index <= lastNodeIndex) {
                 continue;
             }
 
@@ -820,34 +842,34 @@ public class SystemInfoService implements GsonPostProcessable {
                 break;
             }
 
-            long backendId = backend.getId();
-            if (!backendIds.contains(backendId)) {
-                backendIds.add(backendId);
-                lastBackendId = backendId;
+            long nodeId = node.getId();
+            if (!nodeIds.contains(nodeId)) {
+                nodeIds.add(nodeId);
+                lastNodeId = nodeId;
             } else {
                 failed = true;
                 break;
             }
         }
 
-        if (backendIds.size() != backendNum) {
+        if (nodeIds.size() != nodeNum) {
             failed = true;
         }
 
         if (failed) {
             // debug: print backend info when the selection failed
-            for (Backend backend : backends) {
-                LOG.debug("random select: {}", backend);
+            for (ComputeNode node : nodes) {
+                LOG.debug("random select: {}", node);
             }
             return Collections.emptyList();
         }
 
         if (isCreate) {
-            lastBackendIdForCreation = lastBackendId;
+            lastNodeIdForCreation = lastNodeId;
         } else {
-            lastBackendIdForOther = lastBackendId;
+            lastNodeIdForOther = lastNodeId;
         }
-        return backendIds;
+        return nodeIds;
     }
 
     public ImmutableMap<Long, Backend> getIdToBackend() {

--- a/fe/fe-core/src/test/java/com/starrocks/cluster/SystemInfoServiceTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/cluster/SystemInfoServiceTest.java
@@ -49,6 +49,7 @@ import com.starrocks.sql.ast.AddBackendClause;
 import com.starrocks.sql.ast.AlterSystemStmt;
 import com.starrocks.sql.ast.DropBackendClause;
 import com.starrocks.system.Backend;
+import com.starrocks.system.ComputeNode;
 import com.starrocks.system.SystemInfoService;
 import mockit.Expectations;
 import mockit.Mock;
@@ -65,6 +66,7 @@ import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.util.List;
 
 public class SystemInfoServiceTest {
 
@@ -370,6 +372,32 @@ public class SystemInfoServiceTest {
         dis.close();
 
         deleteDir(dir);
+    }
+
+    @Test
+    public void testSeqChooseComputeNodes() {
+        clearAllBackend();
+        AddBackendClause stmt = new AddBackendClause(Lists.newArrayList("192.168.0.1:1234"));
+        com.starrocks.sql.analyzer.Analyzer.analyze(new AlterSystemStmt(stmt), new ConnectContext(null));
+
+        try {
+            GlobalStateMgr.getCurrentSystemInfo().addComputeNodes(stmt.getHostPortPairs());
+        } catch (DdlException e) {
+            Assert.fail();
+        }
+
+        Assert.assertNotNull(GlobalStateMgr.getCurrentSystemInfo().
+                getComputeNodeWithHeartbeatPort("192.168.0.1", 1234));
+
+        List<Long> longList = GlobalStateMgr.getCurrentSystemInfo().seqChooseComputeNodes(1, false, false);
+        Assert.assertEquals(1, longList.size());
+        ComputeNode computeNode = new ComputeNode();
+        computeNode.setHost("192.168.0.1");
+        computeNode.setHttpPort(9030);
+        computeNode.setAlive(true);
+        GlobalStateMgr.getCurrentSystemInfo().addComputeNode(computeNode);
+        List<Long> computeNods = GlobalStateMgr.getCurrentSystemInfo().seqChooseComputeNodes(1, true, false);
+        Assert.assertEquals(1, computeNods.size());
     }
 
 }

--- a/fe/fe-core/src/test/java/com/starrocks/http/StarRocksHttpTestCase.java
+++ b/fe/fe-core/src/test/java/com/starrocks/http/StarRocksHttpTestCase.java
@@ -128,6 +128,15 @@ public abstract class StarRocksHttpTestCase {
     @Mocked
     private static EditLog editLog;
 
+    public static int detectUsableSocketPort() {
+        try (ServerSocket socket = new ServerSocket(0)) {
+            socket.setReuseAddress(true);
+            return socket.getLocalPort();
+        } catch (Exception e) {
+            throw new IllegalStateException("Could not find a free TCP/IP port");
+        }
+    }
+
     public static OlapTable newTable(String name) {
         GlobalStateMgr.getCurrentInvertedIndex().clear();
         Column k1 = new Column("k1", Type.BIGINT);

--- a/fe/fe-core/src/test/java/com/starrocks/http/TransactionLoadActionOnSharedDataClusterTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/http/TransactionLoadActionOnSharedDataClusterTest.java
@@ -15,13 +15,12 @@
 package com.starrocks.http;
 
 
-import com.google.common.collect.ImmutableMap;
-import com.starrocks.catalog.DiskInfo;
 import com.starrocks.common.DdlException;
 import com.starrocks.http.rest.TransactionLoadAction;
 import com.starrocks.http.rest.TransactionResult;
 import com.starrocks.server.GlobalStateMgr;
-import com.starrocks.system.Backend;
+import com.starrocks.server.RunMode;
+import com.starrocks.system.ComputeNode;
 import io.netty.handler.codec.http.HttpMethod;
 import io.netty.handler.codec.http.HttpResponseStatus;
 import mockit.Mock;
@@ -40,19 +39,27 @@ import org.junit.Test;
 
 import java.io.IOException;
 
-public class TransactionLoadActionTest extends StarRocksHttpTestCase {
+public class TransactionLoadActionOnSharedDataClusterTest extends StarRocksHttpTestCase {
 
     private static HttpServer beServer;
 
     @Override
     @Before
     public void setUp() {
-        Backend backend4 = new Backend(1234, "localhost", 8040);
-        backend4.setBePort(9300);
-        backend4.setAlive(true);
-        backend4.setHttpPort(9737);
-        backend4.setDisks(new ImmutableMap.Builder<String, DiskInfo>().put("1", new DiskInfo("")).build());
-        GlobalStateMgr.getCurrentSystemInfo().addBackend(backend4);
+        new MockUp<RunMode>() {
+            @Mock
+            public RunMode getCurrentRunMode() {
+                return RunMode.SHARED_DATA;
+            }
+        };
+
+
+
+        ComputeNode computeNode = new ComputeNode(1234, "localhost", 8040);
+        computeNode.setBePort(9300);
+        computeNode.setAlive(true);
+        computeNode.setHttpPort(9737);
+        GlobalStateMgr.getCurrentSystemInfo().addComputeNode(computeNode);
         new MockUp<GlobalStateMgr>() {
             @Mock
             boolean isLeader() {
@@ -60,10 +67,10 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
             }
         };
     }
-    
+
     @After
     public void tearDown() {
-        GlobalStateMgr.getCurrentSystemInfo().dropBackend(new Backend(1234, "localhost", HTTP_PORT));
+        GlobalStateMgr.getCurrentSystemInfo().dropComputeNode(new ComputeNode(1234, "localhost", HTTP_PORT));
     }
 
     @BeforeClass
@@ -138,7 +145,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
                     @Override
                     public void writeTo(BufferedSink arg0) throws IOException {
                     }
-                    
+
                 })
                 .build();
 
@@ -161,7 +168,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
                     @Override
                     public void writeTo(BufferedSink arg0) throws IOException {
                     }
-                    
+
                 })
                 .build();
 
@@ -185,7 +192,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
                     @Override
                     public void writeTo(BufferedSink arg0) throws IOException {
                     }
-                    
+
                 })
                 .build();
 
@@ -211,7 +218,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
                     @Override
                     public void writeTo(BufferedSink arg0) throws IOException {
                     }
-                    
+
                 })
                 .build();
 
@@ -234,7 +241,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
                     @Override
                     public void writeTo(BufferedSink arg0) throws IOException {
                     }
-                    
+
                 })
                 .build();
 
@@ -261,7 +268,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
                     @Override
                     public void writeTo(BufferedSink arg0) throws IOException {
                     }
-                    
+
                 })
                 .build();
 
@@ -284,7 +291,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
                     @Override
                     public void writeTo(BufferedSink arg0) throws IOException {
                     }
-                    
+
                 })
                 .build();
 
@@ -307,7 +314,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
                     @Override
                     public void writeTo(BufferedSink arg0) throws IOException {
                     }
-                    
+
                 })
                 .build();
 
@@ -334,7 +341,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
                     @Override
                     public void writeTo(BufferedSink arg0) throws IOException {
                     }
-                    
+
                 })
                 .build();
 
@@ -344,4 +351,5 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
 
         Assert.assertEquals(false, res.contains("OK"));
     }
+
 }

--- a/fe/fe-core/src/test/java/com/starrocks/http/TransactionLoadActionOnSharedDataClusterTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/http/TransactionLoadActionOnSharedDataClusterTest.java
@@ -42,6 +42,7 @@ import java.io.IOException;
 public class TransactionLoadActionOnSharedDataClusterTest extends StarRocksHttpTestCase {
 
     private static HttpServer beServer;
+    private static int TEST_HTTP_PORT = 0;
 
     @Override
     @Before
@@ -58,7 +59,7 @@ public class TransactionLoadActionOnSharedDataClusterTest extends StarRocksHttpT
         ComputeNode computeNode = new ComputeNode(1234, "localhost", 8040);
         computeNode.setBePort(9300);
         computeNode.setAlive(true);
-        computeNode.setHttpPort(9737);
+        computeNode.setHttpPort(TEST_HTTP_PORT);
         GlobalStateMgr.getCurrentSystemInfo().addComputeNode(computeNode);
         new MockUp<GlobalStateMgr>() {
             @Mock
@@ -75,7 +76,8 @@ public class TransactionLoadActionOnSharedDataClusterTest extends StarRocksHttpT
 
     @BeforeClass
     public static void initBeServer() throws IllegalArgException, InterruptedException {
-        beServer = new HttpServer(9737);
+        TEST_HTTP_PORT = detectUsableSocketPort();
+        beServer = new HttpServer(TEST_HTTP_PORT);
         BaseAction ac = new BaseAction(beServer.getController()) {
 
             @Override

--- a/fe/fe-core/src/test/java/com/starrocks/http/TransactionLoadActionTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/http/TransactionLoadActionTest.java
@@ -43,6 +43,7 @@ import java.io.IOException;
 public class TransactionLoadActionTest extends StarRocksHttpTestCase {
 
     private static HttpServer beServer;
+    private static int TEST_HTTP_PORT = 0;
 
     @Override
     @Before
@@ -50,7 +51,7 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
         Backend backend4 = new Backend(1234, "localhost", 8040);
         backend4.setBePort(9300);
         backend4.setAlive(true);
-        backend4.setHttpPort(9737);
+        backend4.setHttpPort(TEST_HTTP_PORT);
         backend4.setDisks(new ImmutableMap.Builder<String, DiskInfo>().put("1", new DiskInfo("")).build());
         GlobalStateMgr.getCurrentSystemInfo().addBackend(backend4);
         new MockUp<GlobalStateMgr>() {
@@ -68,7 +69,9 @@ public class TransactionLoadActionTest extends StarRocksHttpTestCase {
 
     @BeforeClass
     public static void initBeServer() throws IllegalArgException, InterruptedException {
-        beServer = new HttpServer(9737);
+        TEST_HTTP_PORT = detectUsableSocketPort();
+
+        beServer = new HttpServer(TEST_HTTP_PORT);
         BaseAction ac = new BaseAction(beServer.getController()) {
 
             @Override

--- a/fe/fe-core/src/test/java/com/starrocks/qe/scheduler/SchedulerTestNoneDBBase.java
+++ b/fe/fe-core/src/test/java/com/starrocks/qe/scheduler/SchedulerTestNoneDBBase.java
@@ -117,7 +117,7 @@ public class SchedulerTestNoneDBBase extends PlanTestNoneDBBase {
 
     public static void makeCreateTabletStable() {
         resetCatalogIdGenerator();
-        resetChooseBackendIds();
+        resetChooseNodeIds();
     }
 
     public static void setBackendService(PBackendService backendService) {
@@ -187,21 +187,21 @@ public class SchedulerTestNoneDBBase extends PlanTestNoneDBBase {
     }
 
     /**
-     * Mock {@link SystemInfoService#seqChooseBackendIds(int, boolean, List)}.
+     * Mock {@link SystemInfoService#seqChooseNodeIds(int, boolean, List)}.
      */
-    private static void resetChooseBackendIds() {
-        AtomicInteger nextBackendIndex = new AtomicInteger(0);
+    private static void resetChooseNodeIds() {
+        AtomicInteger nextNodeIndex = new AtomicInteger(0);
         new MockUp<SystemInfoService>() {
             @Mock
-            public synchronized List<Long> seqChooseBackendIds(int backendNum, boolean isCreate,
-                                                               final List<Backend> srcBackends) {
-                List<Long> backendIds = new ArrayList<>(backendNum);
-                for (int i = 0; i < backendNum; i++) {
-                    int index = nextBackendIndex.getAndIncrement();
-                    long id = srcBackends.get(index % srcBackends.size()).getId();
-                    backendIds.add(id);
+            public synchronized List<Long> seqChooseNodeIds(int nodeNum, boolean isCreate,
+                                                               final List<Backend> srcNodes) {
+                List<Long> nodeIds = new ArrayList<>(nodeNum);
+                for (int i = 0; i < nodeNum; i++) {
+                    int index = nextNodeIndex.getAndIncrement();
+                    long id = srcNodes.get(index % srcNodes.size()).getId();
+                    nodeIds.add(id);
                 }
-                return backendIds;
+                return nodeIds;
             }
         };
     }

